### ISSUE TITLE
react/no-direct-mutation-state: `this.state` mutations with a computed (bracket) member access in the assignment target are not reported

### ISSUE THREAD

**Issue by ahmedelgabri** [oxc-project/oxc#25625]
## Summary

`react/no-direct-mutation-state` fails to flag a direct `this.state` mutation when the left-hand side of the assignment contains a computed (bracket) member access such as `this.state.items[id].prop = ...`. Static-only member chains are detected; any `[...]` access in the chain silently disables detection. The detector appears to stop walking the member-access chain at the first computed access.

## oxlint version

1.78.0; originally observed on 1.69.0, re-verified on 1.72.0 and 1.78.0

## Reproduction

`repro.tsx`:

```tsx
import {Component} from 'react'

class Example extends Component {
	a() {
		this.state.x = 1
	}
	b() {
		this.state.x.y = 1
	}
	c(id: string) {
		this.state.x[id] = 1
	}
	d(id: string) {
		this.state.x[id].y = 1
	}
	e(id: string) {
		this.state.items[id].prop = {}
	}
}
```

`.oxlintrc.json`:

```json
{
	"plugins": ["react"],
	"categories": {"correctness": "off"},
	"rules": {"react/no-direct-mutation-state": "error"}
}
```

Run:

```
oxlint -c .oxlintrc.json --disable-nested-config repro.tsx
```

## Expected

All five assignments (lines 5, 8, 11, 14, 17) are direct mutations of `this.state` and should be flagged. ESLint's `eslint-plugin-react` (the upstream implementation) reports all five: `Do not mutate state directly. Use setState()`.

## Actual

oxlint's native rule reports only the two static chains; the three cases with a computed `[...]` member access in the LHS chain are not reported:

```
repro.tsx:5:3: error react(no-direct-mutation-state): Never mutate `this.state` directly. help: Calling `setState()` afterwards will replace the mutations you made via `this.state`.
repro.tsx:8:3: error react(no-direct-mutation-state): Never mutate `this.state` directly. help: Calling `setState()` afterwards will replace the mutations you made via `this.state`.
```

Knock-on effect: an `// eslint-disable-next-line react/no-direct-mutation-state` placed over such a mutation is reported as `Unused eslint-disable directive (no problems were reported)` under `--report-unused-disable-directives`, because the rule it suppresses never fires — masking a genuine bug-suppression and breaking unused-directive accounting.

## Cross-checked

| Linter | Result on `repro.tsx` |
| --- | --- |
| ESLint 7.23.0 + eslint-plugin-react 7.37.5 (@typescript-eslint/parser 6.21.0) | all five assignments flagged |
| ESLint 9.39.4 flat config + eslint-plugin-react 7.37.5 (typescript-eslint 8.62.1) | all five assignments flagged |
| oxlint 1.78.0 (native rule) | only lines 5 and 8 (static chains) |

This is a correctness gap, not a style issue — the rule's whole purpose is catching accidental direct state mutation, and computed-key writes (`this.state.collection[key] = ...`) are one of the most common ways this bug occurs in real class components.
